src/ripple/shamap/impl/SHAMapInnerNode.cpp

```diff
@@ -176,42 +872,33 @@ SHAMapInnerNode::serializeWithPrefix(Serializer& s) const
     assert(!isEmpty());
 
     s.add32(HashPrefix::innerNode);
-    for (auto const& hh : mHashes)
-        s.addBitString(hh.as_uint256());
+    iterChildren(
+        [&](SHAMapHash const& hh) { s.addBitString(hh.as_uint256()); });
 }
 
 bool
 SHAMapInnerNode::isEmpty() const
 {
-    return mIsBranch == 0;
+    return isBranch_ == 0;
 }
 
 int
 SHAMapInnerNode::getBranchCount() const
 {
-    int count = 0;
-
-    for (int i = 0; i < 16; ++i)
-        if (!isEmptyBranch(i))
-            ++count;
-
-    return count;
+    return popcnt16(isBranch_);
 }
 
 std::string
 SHAMapInnerNode::getString(const SHAMapNodeID& id) const
 {
     std::string ret = SHAMapAbstractNode::getString(id);
-    for (int i = 0; i < mHashes.size(); ++i)
-    {
-        if (!isEmptyBranch(i))
-        {
-            ret += "\n";
-            ret += std::to_string(i);
-            ret += " = ";
-            ret += to_string(mHashes[i]);
-        }
-    }
+    auto hashes = hashesAndChildren_.getHashes();
+    iterNonEmptyChildIndexes([&](auto branchNum, auto indexNum) {
+        ret += "\nb";
```

Comment:
Is the `b` here intentional?

Reply:
Yes. `b` for branch (I assume). So it will print at `b0 = XXX` `b1 = YYY` instead of `0 = XXX`. Note, I did not write that, I just moved the code over. That's been in the code base since 2012 at least.

Reply:
Huh, you're right. In the diff, it matched up against something else, so I didn't spot the 'b'. I'm fine with it.